examples/tilted_bottom_boundary_layer.jl

```diff
@@ -0,0 +1,189 @@
+# stratification as a `BackgroundField` and solve for the buoyancy perturbation. This avoids mean
+# horizontal buoyant gradients, which cannot exist in a horizontally-periodic domain:
+
+b∞(x, y, z, t, p) = p.N²∞ * (x * sin(p.θ_rad) + z * cos(p.θ_rad))
+B_field = BackgroundField(b∞, parameters=(; params.N²∞, params.θ_rad))
+
+
+# ## Bottom drag
+#
+# We also set-up a bottom drag that follows Monin-Obukhov theory, which can be implemented following https://doi.org/10.1029/2005WR004685.
+# Note that we need to include`V∞` in the velocity, since we will model it as a background field.
+#
+
+const κ = 0.4 # von Karman constant
+z₁ = CUDA.@allowscalar znodes(Center, grid)[1] # Closest grid center to the bottom
+cᴰ = (κ / log(z₁/params.z₀))^2 # Drag coefficient
+
+@inline drag_u(x, y, t, u, v, p) = - p.cᴰ * √(u^2 + (v + p.V∞)^2) * u
+@inline drag_v(x, y, t, u, v, p) = - p.cᴰ * √(u^2 + (v + p.V∞)^2) * (v + p.V∞)
+
+drag_bc_u = FluxBoundaryCondition(drag_u, field_dependencies=(:u, :v), parameters=(cᴰ=cᴰ, V∞=params.V∞))
+drag_bc_v = FluxBoundaryCondition(drag_v, field_dependencies=(:u, :v), parameters=(cᴰ=cᴰ, V∞=params.V∞))
+
+u_bcs = FieldBoundaryConditions(bottom = drag_bc_u)
+v_bcs = FieldBoundaryConditions(bottom = drag_bc_v)
+
+V_bg(x, y, z, t, p) = p.V∞
+V_field = BackgroundField(V_bg, parameters=(; V∞=params.V∞))
+#-----
+
+
+
+# ## Create the `NonhydrostaticModel`
+#
+# We create model with a `UpwindBiasedFifthOrder` advection scheme and a `RungeKutta3` timestepper. For simplicity,
+# we use a constant-diffusivity turbulence closure:
+
+model = NonhydrostaticModel(grid = grid, timestepper = :RungeKutta3,
+                            architecture = arch,
+                            advection = UpwindBiasedFifthOrder(),
+                            buoyancy = buoyancy,
+                            coriolis = ConstantCartesianCoriolis(f=params.f₀, rotation_axis=ĝ),
+                            tracers = :b,
+                            closure = IsotropicDiffusivity(ν=params.ν, κ=params.ν),
+                            boundary_conditions = (u=u_bcs, v=v_bcs,),
+                            background_fields = (b=B_field, v=V_field,),
+                           )
+
+
+# ## Create a simulation
+#
+#
+
+using Oceananigans.Grids: min_Δz
+if ndims==3
+    cfl=0.85
+else
+    cfl=0.5
+end
+wizard = TimeStepWizard(max_change=1.03, cfl=cfl)
+
+# Print a progress message
+progress_message(sim) =
+    @printf("i: %04d, t: %s, Δt: %s, wmax = %.1e ms⁻¹, wall time: %s\n",
+            iteration(sim), prettytime(time(sim)),
+            prettytime(sim.Δt), maximum(abs, sim.model.velocities.w),
+            prettytime((time_ns() - start_time) * 1e-9))
+
+simulation = Simulation(model, Δt=0.5*min_Δz(grid)/params.V∞, 
+                        stop_time=3days, 
+                       )
+simulation.callbacks[:wizard] = Callback(wizard, IterationInterval(4))
+simulation.callbacks[:progress] = Callback(progress_message, IterationInterval(10))
```

Comment:
```suggestion
simulation.callbacks[:progress] = Callback(progress_message, IterationInterval(100))
```

The example runs for 1740 iterations now, so an interval of 10 produces 174 lines of output which is probably too much.